### PR TITLE
Sinatra tracer gem

### PR DESCRIPTION
**Pull request by achandras**
This uses the Rack::Tracer gem to handle route tracing.

View rendering is traced by patching the `render` method.

There's probably room for more detailed tracing if desired, but routes and rendering seemed like the two most important operations.

### CODE REVIEW

**Comment by keitwb** [comment]
Add test for spans generated when render is called.

**Comment by achandras** [comment]
@keitwb I'll rename the repos per your suggestion after merging these PRs

File: lib/sinatra/tracer.rb [hunk 1]
```diff
@@ -1,7 +1,39 @@
...
+      end
+
+      def patch_render
+        ::Sinatra::Base.module_eval do
+          alias_method :render_original, :render
+
+          def render(engine, data, options = {}, locals = {}, &block)
+            result = ""
+
+            OpenTracing.global_tracer.start_active_span("Rendering View") do |scope|
```
Comment (mpetazzoni) [inline lib/sinatra/tracer.rb]: Avoid spaces in span names. How about `sinatra.render`?

File: lib/sinatra/tracer.rb [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+              result = render_original(engine, data, options, locals, &block)
+              scope.span.set_tag("sinatra.template", data)
```
Comment (mpetazzoni) [inline lib/sinatra/tracer.rb:27]: What's `data` here?

**Review by mpetazzoni — CHANGES_REQUESTED** [review]

File: lib/sinatra/tracer.rb (hunk 2 above)
Comment (achandras) [reply lib/sinatra/tracer.rb:27]: `data` is the name of the template or the template code itself if the caller is rendering a string. I can add an item in the readme to document this tag

**Review by mpetazzoni — APPROVED** [review]
LGTM. I have some concerns about the `data` tag; I think it's fine if it's a template name, but a raw template string might be quite large in some situations.

**Comment by achandras** [comment]
@max Agreed, in practice it should ideally only be small snippets. Do we want to enforce an arbitrary length limit on the tag?

**Comment by mpetazzoni** [comment]
No it's fine we already enforce a limit on span size in the gateway to protect ourselves.

**Pull request merged**